underscore.js

```diff
@@ -436,6 +436,11 @@
   _.toArray = function(obj) {
     if (!obj) return [];
     if (_.isArray(obj)) return slice.call(obj);
+    if (_.isString(obj)) {
+      // Keep surrogate pair characters together
+      // https://mathiasbynens.be/notes/javascript-unicode
+      return obj.match(/[^\ud800-\udbff\udc00-\udfff]|[\ud800-\udbff][\udc00-\udfff]/g);
```

Comment:
I'll break down the regexp:
```js
/[^\ud800-\udbff\udc00-\udfff]|[\ud800-\udbff][\udc00-\udfff]|[\ud800-\udbff\udc00-\udfff]/g
```
I went with a negated character class to keep things simple and reuse the same unicode ranges in the regexp.

`[^\ud800-\udbff\udc00-\udfff]` means match anything that isn't a high or low surrogate, so regular characters like `a` or `@`.

`[\ud800-\udbff][\udc00-\udfff]` these two character classes match a high and low surrogate pair.

`[\ud800-\udbff\udc00-\udfff]` this last one looks similar but matches lone surrogates that are not paired.